### PR TITLE
fix QgsDualEdgeTriangulation

### PR DESCRIPTION
**Pull request by vcloarec**
Another issue in the dual edge triangulation (described here : https://github.com/qgis/QGIS/issues/30549#issuecomment-678065860), and still present with recent fix.
This PR fixes it.

### CODE REVIEW

File: src/analysis/interpolation/qgsdualedgetriangulation.cpp [hunk 1]
```diff
@@ -204,7 +205,22 @@ int QgsDualEdgeTriangulation::addPoint( const QgsPoint &p )
         int point1 = mHalfEdge[mEdgeOutside]->getPoint();
         int point2 = mHalfEdge[mHalfEdge[mEdgeOutside]->getDual()]->getPoint();
         double distance1 = p.distance( *mPointVector[point1] );
+        if ( distance1 <= leftOfTresh ) // point1 == new point
+        {
+          QgsPoint *pt = mPointVector.last();
```
Comment (nyalldawson) [inline src/analysis/interpolation/qgsdualedgetriangulation.cpp:210]:
```suggestion
          delete mPointVector.takeLast();
```
Etc

**Comment by nyalldawson** [comment]
Any chance of a test for this too?

**Comment by vcloarec** [comment]
yes... this could be a good think, I was thinking about...

**Comment by vcloarec** [comment]
new tests added, and another issue fixed...

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**